realm/src/main/java/io/realm/internal/Context.java

```diff
@@ -62,14 +62,14 @@ public void executeDelayedDisposal() {
             }
             abandonedQueries.clear();
 
-            cleanRows();
+            cleanNativeReferences();
         }
     }
 
-    public void cleanRows() {
+    public void cleanNativeReferences() {
         NativeObjectReference reference = (NativeObjectReference) referenceQueue.poll();
         while (reference != null) {
-            UncheckedRow.nativeClose(reference.nativePointer);
+            reference.cleanup();
```

Comment:
the reason of `cleanup` here is it has to be `abstract` in case the subclass forgets to implement it.
But it does look better by calling `clear` here.
I will keep the `cleanup` but make it `protected` and call it from `NativeReference.clear` instead.